File: Python/word_count/Serial/top5overall.py
import os
import sys
sys.path.append('../../')
from word_count_helpers import *

# Load the file names                                                                                                                                      
dataDir = "../../../data/word_count/"
fnames = os.listdir(dataDir)

allcounts = []
for fname in fnames:
    f = open(dataDir+fname, 'r', encoding='utf-8')
    text = cleantext(f.readlines())
    allcounts.append(countwords(text))

globalcounts = dict()
for counts in allcounts:
    globalcounts = { k: counts.get(k, 0) + globalcounts.get(k, 0) for k in set(counts) | set(globalcounts) }
    
top5 = sorted(globalcounts, key=globalcounts.get, reverse=True)[:5]
for k in top5:
    print("{}: {}".format(k, globalcounts[k]))

File: Julia/word_count/Serial/top5overall.jl
# Read in file of helper functions
# This file incluse: cleantext, countwords
include("../word_count_helpers.jl")

# Load the file names
dataLoc = "../../../data/word_count/";
fnames = dataLoc.*readdir(dataLoc)

# Create function for processing each file
function getwordcounts(fname)
    # Read in file and clean the text
    f = open(fname)
    text=cleantext(readlines(f))

    # Count number of times each word appears
    return countwords(text)
end

# Get counts for each book
allcounts = getwordcounts.(fnames)

# Calculate the overall word counts
overallcounts = merge(+,allcounts...)

# Sort and print the top 5 words with their counts
println(sort(collect(overallcounts), by=x->x[2], rev=true)[1:5,:])
